src/Project.ts

```diff
@@ -100,37 +121,38 @@ export default class Project {
     return matchingTriggers;
   }
 
-  _stepEdgeActivatedTriggers() {
-    const edgeActivated = this._matchingTriggers(tr => tr.isEdgeActivated);
+  private _stepEdgeActivatedTriggers(): void {
+    const edgeActivated = this._matchingTriggers((tr) => tr.isEdgeActivated);
     const triggersToStart = [];
     for (const triggerWithTarget of edgeActivated) {
       const { trigger, target } = triggerWithTarget;
       let predicate;
       switch (trigger.trigger) {
         case Trigger.TIMER_GREATER_THAN:
-          predicate = this.timer > trigger.option("VALUE", target);
+          predicate = this.timer > trigger.option("VALUE", target)!;
           break;
         case Trigger.LOUDNESS_GREATER_THAN:
-          predicate = this.loudness > trigger.option("VALUE", target);
+          predicate = this.loudness > trigger.option("VALUE", target)!;
           break;
         default:
-          throw new Error(`Unimplemented trigger ${trigger.trigger}`);
+          throw new Error(`Unimplemented trigger ${String(trigger.trigger)}`);
       }
 
       // Default to false
-      const prevPredicate = !!this._prevStepTriggerPredicates.get(trigger);
-      this._prevStepTriggerPredicates.set(trigger, predicate);
+      const prevPredicate =
+        !!this._prevStepTriggerPredicates.get(triggerWithTarget);
+      this._prevStepTriggerPredicates.set(triggerWithTarget, predicate);
```

Comment:
Is this a runtime/behavioral change? It looks like the value of `prevPredicate` would evaluate differently here, but I'm not versed in edge-activated triggers to be certain. If it's behavioral, should this change be a separate bug fix pull request?

Reply:
Looks like I introduced a bug; good catch!